packages/web/tests/components/DiscordCheck/DiscordButtonCheck.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import fetchMock from 'fetch-mock-jest';
+import userEvent from '@testing-library/user-event';
+import { DiscordButtonCheck } from '../../../src/components/DiscordCheck';
+import { act, render, screen } from '../../testUtils/testTools';
+
+jest.mock('next/router', () => ({
+  useRouter() {
+    return {
+      query: '',
+    };
+  },
+}));
+
+const useRouter = jest.spyOn(require('next/router'), 'useRouter');
+
+interface User {
+  discordId?: string;
+}
+
+const sampleUser: User = {
+  discordId: undefined,
+};
+
+const sampleUser2: User = {
+  discordId: '34523452345',
+};
+
+const wait = () => new Promise<void>((resolve) => setTimeout(() => resolve(), 0));
+
+describe('check for discordId to change button', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it('undefined user', async () => {
+    jest.isolateModules(async () => {
+      useRouter.mockImplementation(() => ({
+        query: undefined,
+      }));
+
+      expect(() => render(<DiscordButtonCheck />)).not.toThrow();
+
+      await act(wait);
+      expect(screen.getByText('User could not be found'));
+    });
+  });
+
+  it('renders join discord button when user has no discord id', async () => {
+    jest.isolateModules(async () => {
+      useRouter.mockImplementation(() => ({
+        query: {},
+      }));
+
+      fetchMock.get('/api/users/me', sampleUser);
+
+      expect(() => render(<DiscordButtonCheck />)).not.toThrow();
+      await act(wait);
+
+      expect(screen.queryByText('Join our Discord')).toBeVisible();
```

Comment:
You're referencing the same element 4 times in this test; create a variable and use it everywhere instead
```suggestion
      const joinDiscordButton = screen.queryByText('Join our Discord')
      expect(joinDiscordButton).toBeVisible();
```